src/controllers/dynakube/activegate/internal/capability/reconciler_test.go

```diff
@@ -1,517 +1,360 @@
-// 		)
-// 	})
-// }
+import (
+	"context"
+	"testing"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/capability"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/internal/authtoken"
+	"github.com/Dynatrace/dynatrace-operator/src/kubesystem"
+	"github.com/Dynatrace/dynatrace-operator/src/scheme"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testToken    = "dt.testtoken.test"
+	testUID      = "test-uid"
+	testDynakube = "test-dynakube"
+)
+
+func createClient() client.WithWatch {
+	return fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithObjects(&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: kubesystem.Namespace,
+				UID:  testUID,
+			},
+		}).
+		WithObjects(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dynatracev1beta1.AuthTokenSecretSuffix,
+				Namespace: testNamespace,
+			},
+			Data: map[string][]byte{authtoken.ActiveGateAuthTokenName: []byte(testToken)},
+		}).
+		Build()
+}
+
+func createInstance() *dynatracev1beta1.DynaKube {
+	instance := &dynatracev1beta1.DynaKube{
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			ActiveGate: dynatracev1beta1.ActiveGateSpec{
+				Capabilities: []dynatracev1beta1.CapabilityDisplayName{
+					dynatracev1beta1.RoutingCapability.DisplayName,
+					dynatracev1beta1.KubeMonCapability.DisplayName,
+					dynatracev1beta1.MetricsIngestCapability.DisplayName,
+					dynatracev1beta1.DynatraceApiCapability.DisplayName,
+					dynatracev1beta1.SyntheticCapability.DisplayName,
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testDynakube,
+			Namespace: testNamespace,
+		}}
+	return instance
+}
+
+func createInstanceWithoutActiveGateService() *dynatracev1beta1.DynaKube {
+	instance := &dynatracev1beta1.DynaKube{
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			ActiveGate: dynatracev1beta1.ActiveGateSpec{
+				Capabilities: []dynatracev1beta1.CapabilityDisplayName{
+					dynatracev1beta1.KubeMonCapability.DisplayName,
+					dynatracev1beta1.SyntheticCapability.DisplayName,
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testDynakube,
+			Namespace: testNamespace,
+		}}
+	return instance
+}
+
+func getMockReconciler() *MockReconciler {
+	mockReconciler := MockReconciler{}
+	mockReconciler.On("Reconcile").Return(nil)
+	return &mockReconciler
+}
+
+func verifyReconciler(t *testing.T, r *Reconciler) {
+	require.NotNil(t, r)
+	require.NotNil(t, r.client)
+	require.NotNil(t, r)
+	require.NotNil(t, r.dynakube)
+}
+
+func TestReconcile(t *testing.T) {
+	t.Run(`Reconciler works with multiple capabilities`, func(t *testing.T) {
+		clt := createClient()
```

Comment:
since most of these tests reuse the same 3/4 variables, is it possible to define them before the `t.Run` or does that cause issues with writing to the same variable?

Reply:
thank you, i did that everywhere where it is possible :+1: